server/interpreter.js

```diff
@@ -2900,6 +2661,15 @@ Interpreter.prototype['stepForInStatement'] = function() {
             }
           }
         }
+      } else {
+        for (var prop in state.object_) {
```

Comment:
Is this case to deal with the case of `for (x in "foo")`?  If so, it is wrong, because it ends up iterating over the properties of `String.prototype` instead of `Interpreter.STRING.prototype` (and similarly for other primitives).

(This is one case where "destructive" testing—i.e., the sort you wouldn't want to do on a running production core—can be very useful.)